scanspec/service.py

```diff
@@ -62,12 +62,30 @@ class AxisFrames:
     """A fixed reference that can be scanned. i.e. a motor, time or
     number of repetitions.
     """
-    lower: Optional[Points]
+    lower: Points
     """The lower bounds of each midpoint (used when fly scanning)"""
-    midpoints: Optional[Points]
-    """The centre points of the scan"""
-    upper: Optional[Points]
+    midpoints: Points
+    """The middle points of the scan"""
+    upper: Points
     """The upper bounds of each midpoint (used when fly scanning)"""
+    smallest_step: float
+    """The smallest step between midpoints in this axis"""
+
+
+@dataclass
+class SmallestStep:
+    """ The smallest distance between midpoints in a multidimensional scan space"""
+
+    def __init__(self, points: List[ndarray]):
```

Comment:
I wonder if it makes sense to pass the whole `AxesPoints` dictionary here instead?

Reply:
I would argue this whole class should be a resolver on `PointsResponse`. This would make `PointsResponse` into a class like `Points` that was passed the whole chunk, then only calculated `AxisFrames` or `smallest_abs_step` on demand. `AxisFrames` would probably look similar...

Reply:
This was a great suggestion. Not only should it be much more performant, but it also reduces the number of fields you need to request in graphql.